app/models/index.rb

```diff
@@ -20,6 +20,8 @@ def publish!(at = Time.zone.now)
     update(published_at: at)
   end
 
+  scope :user_indices, ->(user) { where(user_id: user) }
```

Comment:
Not needed, you should be able to do the same thing using current_user.indices relation